test/releasers/java-backport.ts

```diff
@@ -0,0 +1,630 @@
+    });
+
+    sandbox
+      .stub(releasePR.gh, 'getRepositoryDefaultBranch')
+      .returns(Promise.resolve('master'));
+
+    // No open release PRs, so create a new release PR
+    sandbox
+      .stub(releasePR.gh, 'findOpenReleasePRs')
+      .returns(Promise.resolve([]));
+
+    sandbox
+      .stub(releasePR.gh, 'findMergedReleasePR')
+      .returns(Promise.resolve(undefined));
+
+    // Indicates that there are no PRs currently waiting to be released:
+    sandbox.stub(releasePR, 'latestTag').returns(
+      Promise.resolve({
+        name: 'v0.20.3',
+        sha: 'abc123',
+        version: '0.20.3',
+      })
+    );
+
+    const findFilesStub = sandbox.stub(
+      releasePR.gh,
+      'findFilesByFilenameAndRef'
+    );
+    findFilesStub
+      .withArgs('pom.xml', defaultBranch, undefined)
+      .resolves(['pom.xml']);
+    findFilesStub
+      .withArgs('build.gradle', defaultBranch, undefined)
+      .resolves([]);
+    findFilesStub
+      .withArgs('dependencies.properties', defaultBranch, undefined)
+      .resolves([]);
+
+    const getFileContentsStub = sandbox.stub(
+      releasePR.gh,
+      'getFileContentsOnBranch'
+    );
+    getFileContentsStub
+      .withArgs('versions.txt', defaultBranch)
+      .resolves(buildFileContent('versions-lts-snapshot.txt'));
+    getFileContentsStub
+      .withArgs('README.md', defaultBranch)
+      .resolves(buildFileContent('README.md'));
+    getFileContentsStub
+      .withArgs('pom.xml', defaultBranch)
+      .resolves(buildFileContent('pom.xml'));
+    getFileContentsStub
+      .withArgs(
+        'google-api-client/src/main/java/com/google/api/client/googleapis/GoogleUtils.java',
+        defaultBranch
+      )
+      .resolves(buildFileContent('GoogleUtils.java'));
+    getFileContentsStub.rejects(
+      Object.assign(Error('not found'), {status: 404})
+    );
+
+    sandbox
+      .stub(releasePR.gh, 'commitsSinceSha')
+      .resolves([buildMockCommit('chore: something irrelevant')]);
+
+    // should not attempt to create a pull request
+    sandbox
+      .stub(suggester, 'createPullRequest')
+      .rejects(Error('should not get here'));
+
+    await releasePR.run();
+  });
+
+  describe('latestTag', () => {
+    let req: nock.Scope;
+    let releasePR: ReleasePR;
+
+    beforeEach(() => {
+      req = nock('https://api.github.com/');
+
+      releasePR = new JavaBackport({
+        github: new GitHub({owner: 'googleapis', repo: 'java-trace'}),
+      });
+
+      sandbox.stub(releasePR.gh, 'getRepositoryDefaultBranch').resolves('main');
+    });
+
+    it('returns a stable branch pull request', async () => {
+      const graphql = JSON.parse(
+        readFileSync(
+          resolve('./test/fixtures', 'latest-tag-sp-version.json'),
+          'utf8'
+        )
+      );
+      req.post('/graphql').reply(200, {
+        data: graphql,
+      });
+      const latestTag = await releasePR.latestTag();
+      expect(latestTag!.version).to.equal('1.127.0-sp.1');
```

Comment:
Can we use a version without "-sp.1" for test data?

Reply:
I will just remove this test - it was copied from the LTS test. The purpose of this one was to detect the special -sp.1 version scheme which is irrelevant to this releaser.